### PR TITLE
jakub to dev

### PR DESCRIPTION
**Pull request by ZaiZu2**

### CODE REVIEW

File: dbhydra/dbhydra_core.py [hunk 1]
```diff
@@ -468,12 +470,17 @@ class Mysqldb(AbstractDB):
     'datetime': "datetime"
     }
 
+    def connect_to_db(self):
+        self.connection = MySQLdb.connect(host=self.DB_SERVER, port=self.DB_PORT, user=self.DB_USERNAME, password=self.DB_PASSWORD, database=self.DB_DATABASE)
+        self.cursor = self.connection.cursor()
```
Comment (Omastto1) [inline dbhydra/dbhydra_core.py:475]: there is `connect_to_db` implemented in `AbstractDB` that calls this method:
```    
def _connect(self):
        if self.locally:
            self.connect_locally()
        else:
            self.connect_remotely()
```
so this one seems extra

File: dbhydra/dbhydra_core.py [hunk 2]
```diff
@@ -763,7 +770,7 @@ def update(self, variable_assign, where=None):
         else:
             query = "UPDATE " + self.name + " SET " + variable_assign + " WHERE " + where
         print(query)
-        self.db1.execute(query)
+        return self.db1.execute(query)
```
Comment (Omastto1) [inline dbhydra/dbhydra_core.py:773]: it is not clear to me what the returned value should be?

File: dbhydra/dbhydra_core.py [hunk 3]
```diff
@@ -1772,17 +1786,15 @@ def dict_to_df(dictionary, column1, column2):
     df = pd.DataFrame(list(dictionary.items()), columns=[column1, column2])
     return (df)
 
-from pydantic import BaseModel
-from typing import List, Dict
 
 class AbstractModel(abc.ABC, BaseModel):
```
Comment (DovaX) [inline dbhydra/dbhydra_core.py:1790]: check

**Review by DovaX — APPROVED** [review]

File: dbhydra/dbhydra_core.py (hunk 1 above)
Comment (ZaiZu2) [reply dbhydra/dbhydra_core.py:475]: > there is `connect_to_db` implemented in `AbstractDB` that calls this method:
> 
> ```
> def _connect(self):
>         if self.locally:
>             self.connect_locally()
>         else:
>             self.connect_remotely()
> ```
> 
> so this one seems extra

`connect_to_db` is implemented as a context manager, and called as a normal method within the `AbstractDB` init method. To my understanding, it did not make much sense (and doesn't work), so I decided to quickly hotfix it by overwriting this method in the child.

`    @contextmanager
    def connect_to_db(self):
        try:
            self._connect()
            yield None
        finally:
            self.close_connection()
`

File: dbhydra/dbhydra_core.py (hunk 2 above)
Comment (ZaiZu2) [reply dbhydra/dbhydra_core.py:773]: For some SQL DBs, when a statement is executed successfully, DB can return some information. E.g. for INSERT, DB will return a PK id under which the record was inserted.

File: dbhydra/dbhydra_core.py (hunk 3 above)
Comment (ZaiZu2) [reply dbhydra/dbhydra_core.py:1790]: Imports shifted to the top as a part of a small clean up.

**Pull request merged**